File: portscaner.rb
require 'socket'
def port_scanner(ip, ports)
    puts "Verificando portas no IP: #{ip}"

    ports.each do |port|
        begin
            socket = TCPSocket.new(ip, port)
            puts "Porta #{port}: ABERTA"

        rescue Errno: :ECONNREFUSED
            puts"Porta #{port}: FECHADA"

        rescue Errno: :ETIMEDOUT
            puts "Porta #{port}: SEM RESPOSTA"

        ensure
            socket.close is socket 

        end
    end
end

puts "Digite o  ip  ou Hostname  a ser escaneado:"
ip = gets.chomp

ports = (1..1024).to_a
port_scanner(ip, ports)

File: portscaner.py
import socket

def port_scanner(ip, ports):
    print(f"Verificando  portas no  IP: {ip}")
    
    for port in ports:
        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
            s.settimeout(1)
            
            result = s.connect_ex((ip, port))
            
            if result == 0:
                print(f"Porta {port}: ABERTA")
            else: 
                print(f"Porta {port}: FECHADA")

ip = input ("Digite o IP ou hostname a ser escaneado: ")            

ports = range (1, 1025)

port_scanner(ip, ports)